lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -116,25 +129,30 @@ newNetworkLayer
     :: forall n. ()
     => BaseUrl
     -> IO (NetworkLayer (Jormungandr n) IO)
-newNetworkLayer url = do
+newNetworkLayer = fmap snd . newNetworkLayer'
+
+-- | Creates a new 'NetworkLayer' connecting to an underlying 'Jormungandr'
+-- backend target. Also returns the internal 'JormungandrLayer' component.
+newNetworkLayer'
+    :: forall n. ()
+    => BaseUrl
+    -> IO (JormungandrLayer n IO, NetworkLayer (Jormungandr n) IO)
+newNetworkLayer' url = do
```

Comment:
Hmmm ... Let's ... not :grimacing: 
I assume this (the extra method with a `'`) is to avoid having to modify all calls to `newNetworkLayer` but I'd rather go for that second option instead of increase the size of the API with overlapping methods. Generally speaking, I don't think it's a good idea to expose any two methods with identical names. Most callers (and they aren't that many!) will only care about the right side, and it's fine. 

Yet, curious to know your point of view on this?

Reply:
I am hoping we could soon delete this function. I don't like it because it exposes `JormungandrLayer` and `mkNetworkLayer` which are implementation details (they might be helpful for testing, but the main CLI shouldn't be using them.

Reply:
What do I need to change? Just the name of the function or something else?

Reply:
I just didn't understand the need for this function. I find it rather confusing that it simply exists. From a reader perspective, it's unclear which one should be called and I don't see much value in having both. I'd rather have the `st <- newMVar ...` moved into `mkNetworkLayer` and pretty much leave calls constructing the network layer as they were.
But, no strong feeling on that, it's quite minor and I mainly wanted to get your opinion on the _why_, thanks :)